pkg/executables/cmk.go

```diff
@@ -91,11 +147,16 @@ func (c *Cmk) ValidateServiceOfferingPresent(ctx context.Context, domain string,
 	return nil
 }
 
-// TODO: Add support for domain, account filtering
-func (c *Cmk) ValidateAffinityGroupsPresent(ctx context.Context, domain string, zone v1alpha1.CloudStackResourceRef, account string, affinityGroupIds []string) error {
+func (c *Cmk) ValidateAffinityGroupsPresent(ctx context.Context, domainId string, account string, affinityGroupIds []string) error {
+	var filterArgs []string
 	for _, affinityGroupId := range affinityGroupIds {
-		idFilterParam := fmt.Sprintf("id=\"%s\"", affinityGroupId)
-		result, err := c.exec(ctx, "list", "affinitygroups", idFilterParam)
+		filterArgs = []string{"list", "affinitygroups"}
+		filterArgs = append(filterArgs, fmt.Sprintf("id=\"%s\"", affinityGroupId))
+		if len(domainId) > 0 || len(account) > 0 {
+			filterArgs = append(filterArgs,  fmt.Sprintf("domainid=\"%s\"", domainId))
+			filterArgs = append(filterArgs,  fmt.Sprintf("account=\"%s\"", account))
```

Comment:
I don't think this logic will work. Domain can be defined without an account specified, so we can't just always add the account filter

Reply:
good point. will update with a comment to explain the situation.